wsdl/ver10/schema/onvif.xsd

```diff
@@ -7631,6 +7631,131 @@ and sample rate. </xs:documentation>
 		<xs:anyAttribute processContents="lax"/>
 	</xs:complexType>
 	<!--===============================-->
+	<xs:simpleType name="TargetFormat">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="MP4">
+				<xs:annotation>
+					<xs:documentation>
+						MP4 files with all tracks in a single file.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="CMAF">
+				<xs:annotation>
+					<xs:documentation>
+						CMAF compliant MP4 files with 1 track per file.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+		</xs:restriction>
+	</xs:simpleType>
+	<!--===============================-->
+	<xs:simpleType name="EncryptionMode">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="CENC">
+				<xs:annotation>
+					<xs:documentation>
+						AES-CTR mode full sample and video NAL Subsample encryption, defined in ISO/IEC 23001-7.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="CBCS">
+				<xs:annotation>
+					<xs:documentation>
+						AES-CBC mode partial video NAL pattern encryption, defined in ISO/IEC 23001-7.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+		</xs:restriction>
+	</xs:simpleType>
+	<!--===============================-->
+	<xs:complexType name="RecordingEncryption">
+		<xs:sequence>
+			<xs:element name="KID" type="xs:string">
```

Comment:
Hi Felix,

I was discussing encryption with Maxime today and we realized that we should enforce the KID to be a UUID through the xs:hexBinary type (minLength=32 & maxLength=32) so that devices can use the KID in the ProtectionSystemSpecificHeaderBox (pssh). See section 8.1.2 from ISO/IEC 23001-7)

Reply:
I have already added the line "The device shall interpret the `KID` configured for the encryption entry as a 16-byte hexadecimal value and use this value as the key identifier." to the encryption section in the spec. Is this enough? Talking about possible asymmetric encryption again, would you then want to provide one key ID of the certificate and one key ID for the `pssh` box?

Reply:
The problem we have for using certificate as KID is that certificate KeyID is a NCName generated by the device and NCName forces to start with a letter, which blocks a lot of UUIDs, so I am not certain we can enforce this constraint if we use a string.

Since pssh is already well defined in the spec and I believe most devices will want to use this box to provide the KeyID used for encryption, then I think we should enforce that specification at the config level and rethink how the certificate ID can be provisioned in the storage configuration.

Reply:
I think we can move this discussion to #315 and continue Bologna.